dbt/include/athena/macros/adapters/relation.sql

```diff
@@ -1,16 +1,11 @@
 {% macro athena__drop_relation(relation) -%}
-  {% set rel_type = adapter.get_table_type(relation.schema, relation.table) %}
+  {% set rel_type = relation.type %}
+  {%- set materialized = config.get('materialized') -%}
   {%- if rel_type is not none %}
-    {%- if rel_type == 'table' %}
+    {%- if rel_type == 'table' and materialized != 'table_hive_ha' %}
```

Comment:
Please remove table_hive_ha, it's a bit complicated than that.
table_hive_ha can be dropped in case of full refresh and also we don't have rel_type table_hive_ha